### DISCUSSION TITLE
Omitted when the main menu is not wide enough just like vscode  on window

### DISCUSSION THREAD

**Discussion by lqd1434** [eclipse-theia/theia#13457] [Q&A] [answered]
I found theia now can toggle main menu and let it in the top of side panel,  but I want to know if Theia has plans to support behavior like vscode. vscode will automatically shrink the main menu bar and display an ellipsis when the window is collapsed. Clicking on the ellipsis can display the collapsed menu through the context menu.

I have tried to implement this function myself, but I encountered a problem. All menus are bound to the path `MAIN_MENU_BAR`. I cannot render them through `ContextMenuRender` and specify the path as `MAIN_MENU_BAR`. There will be many unnecessary menu items here. , they are not hidden, so I was wondering if there is a way to remove this part of the menu

**Comment by JonasHelming** [comment]
"vscode will automatically shrink the main menu bar and display an ellipsis when the window is collapsed. Clicking on the ellipsis can display the collapsed menu through the context menu."

=> I tried this and failed to get this behavior. Can you describe this more in detail, please? What exactly means "collapsed"?

**Reply by lqd1434** [reply]
sorry, what I expressed was not very accurate, i just want to say fold ,like the follow image

![image](https://github.com/eclipse-theia/theia/assets/66897781/a61d90e9-2430-4b54-bcc2-b0cfea3fce7e)

**Reply by lqd1434** [reply]
And this is behavior only in windows

**Reply by JonasHelming** [reply]
OK, thank you, now I can see what you are referring to. I am not aware of any plans to implement this behavior, though. Please feel free to open a feature request and of course contributions are welcome!

**Reply by lqd1434** [reply]
Thank you for your reply，and what do you think about  

> "I have tried to implement this function myself, but I encountered a problem. All menus are bound to the path MAIN_MENU_BAR. I cannot render them through ContextMenuRender and specify the path as MAIN_MENU_BAR. There will be many unnecessary menu items here. , they are not hidden, so I was wondering if there is a way to remove this part of the menu"

**Reply by tsmaeder** [reply, marked as answer]
I guess the place where to start looking would be browser-menu-factory, etc. Not sure how to realize such a feature, though. Native menus might be a challenge.

**Reply by lqd1434** [reply]
thanks, i will try it

**Reply by skysefall-1234** [reply]
Hi @lqd1434, did you find any solution, how to integrate an ellipses feacture to the main toolbar menus when resizing(zoom-in) the UI window. Is there any work around, if there is no support provided from theia.
